kore/src/main/scala/org/kframework/definition/outer.scala

```diff
@@ -309,6 +308,45 @@ case class Module(val name: String, val imports: Set[Import], localSentences: Se
     }
     functionRules
   }
+
+
+  val hasAttributesMap: mutable.Map[Att.Key, java.lang.Boolean] = mutable.Map[Att.Key, java.lang.Boolean]()
```

Comment:
One of the benefits of using scala is that it ensures immutability.
I would strongly avoid using mutable classes here.

Reply:
Agree 100%. We should be using the existing attributes, or else a lazy val that computes some information about the module.

Reply:
The problem is that we depend on the result of `RuleInfo` and `ExpandMacros` to know whether we should have an attribute on that list. Also, we depend on the `heatCoolEq` value.

Reply:
These informations comes outside the information available in the module itself.

Reply:
If this information is required in only one step, you can compute it locally.
Pretty much what RuleInfo does already. (IIRC)